src/Sentry.AspNetCore/SentryTracingMiddleware.cs

```diff
@@ -157,6 +157,10 @@ public async Task InvokeAsync(HttpContext context)
                     {
                         transaction.Finish(status);
                     }
+                    else if (status == SpanStatus.Ok)
+                    {
+                        transaction.Finish(exception);
```

Comment:
```suggestion
                    // Status code not yet changed to 500 but an exception does exist
                    // so lets avoid passing the misleading 200 down and close only with
                    // the exception instance that will be inferred as errored
                    else if (status == SpanStatus.Ok && exception is not null)
                    {
                        transaction.Finish(exception);
```

Reply:
Doesn't the previous lines guarantee that you don't have a null exception?
```csharp
                    if (exception is null)
                    {
                        transaction.Finish(status);
                    }
                    else if (status == SpanStatus.Ok)
                    {
                        transaction.Finish(exception);
                    }
                    else
                    {
                        transaction.Finish(exception, status);
                    }
```

Reply:
Slightly edited the test Transaction_binds_exception_thrown for checking if the Span was set as Internal Error.
Without the changes in this PR, the test was returning a span with the Status Ok.

Reply:
Yeah it already checks for null

Reply:
OK, please add the comments